src/app/_components/Form/ControlledFormCheckbox.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  useFormContext,
+  Controller,
+  type FieldValues,
+  type FieldPathByValue,
+} from 'react-hook-form'
+
+import {
+  FormCheckbox,
+  type FormCheckboxProps,
+} from '@/components/Form/FormCheckbox'
+
+type ControlledProps =
+  | 'checked'
+  | 'defaultChecked'
+  | 'value'
+  | 'defaultValue'
+  | 'onChange'
+  | 'onBlur'
+  | 'required'
```

Comment:
Should we exclude 'required'? What if we have a case something like, "you need to agree with the Privacy Policy"

Reply:
Here's how it's currently set up:

For controlled components, the Zod schema (and the `zodResolver` integration) determines whether a field is required. For instance:

```ts 
const TestSchema = z.object({
  isCool: z.boolean({ message: 'This field is required' }),
```

For a privacy policy check, we can use the uncontrolled version of the component, in which case we can pass `required` as a prop.

Does that make sense?